From 2bd2e52380ba9c568ceba2d8d92b9cd50a22c881 Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Tue, 2 Apr 2019 17:02:13 +0200
Subject: [PATCH 1/2] Passthrough %license tag

matters for fwupd package file /usr/share/licenses/fwupd/COPYING

and added 3 more bits from rpm/lib/rpmfiles.h
---
 pesign-gen-repackage-spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/pesign-gen-repackage-spec b/pesign-gen-repackage-spec
index 1856d9d..9cd374a 100755
--- a/pesign-gen-repackage-spec
+++ b/pesign-gen-repackage-spec
@@ -345,6 +345,10 @@ my %filetypes = (
 	missingok => (1 << 3),
 	noreplace => (1 << 4),
 	ghost     => (1 << 6),
+	license   => (1 << 7),
+	readme    => (1 << 8),
+	pubkey    => (1 << 11),
+	artifact  => (1 << 12),
 );
 
 my %verifyflags = (
@@ -381,7 +385,9 @@ sub print_files {
 			}
 			$attrs .= "(" . join(",", @cfg_attrs) . ")" if @cfg_attrs;
 		}
-		$attrs .= "%doc " if $f->{flags} & $filetypes{doc};
+		for my $filetype (qw(doc license readme pubkey artifact)) {
+			$attrs .= "%$filetype " if $f->{flags} & $filetypes{$filetype};
+		}
 		if ($f->{flags} & $filetypes{ghost}) {
 			$attrs .= "%ghost ";
 			if (S_ISREG($f->{mode})) {
-- 
2.21.0

